src/System.Memory/tests/Memory/MemoryManager.cs

```diff
@@ -28,30 +28,14 @@ public static void MemoryFromMemoryManagerInt()
         }
 
         [Fact]
-        public static void MemoryManagerCtorDefault()
+        public static void MemoryManagerMemoryCtorInvalid()
         {
-            MemoryManager<int> managerInt = default;
-            Assert.Throws<ArgumentNullException>(() => new Memory<int>(managerInt, 0, 0));
+            MemoryForInternalMemoryCreate<int> manager = new MemoryForInternalMemoryCreate<int>();
 
-            managerInt = null;
-            Assert.Throws<ArgumentNullException>(() => new Memory<int>(managerInt, 0, 0));
-
-            MemoryManager<object> managerObject = default;
-            Assert.Throws<ArgumentNullException>(() => new Memory<object>(managerObject, 0, 0));
-        }
-
-        [Fact]
-        public static void MemoryManagerCtorInvalid()
-        {
-            int[] a = { 91, 92, -93, 94 };
-            MemoryManager<int> manager = new CustomMemoryForTest<int>(a);
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, 0, -1));
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, -1, 0));
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, -1, -1));
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, -1, -1));
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, 0, a.Length + 1));
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, a.Length + 1, 0));
-            Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, 1, a.Length));
+            Assert.Throws<ArgumentOutOfRangeException>(() => manager.CreateMemoryForTest(-1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => manager.CreateMemoryForTest(0, -1));
+            Assert.Throws<ArgumentOutOfRangeException>(() => manager.CreateMemoryForTest(-1, 0));
+            Assert.Throws<ArgumentOutOfRangeException>(() => manager.CreateMemoryForTest(-1, -1));
```

Comment:
Can we keep all the original test cases, including the ones where we pass in start/length beyond the array length?

```C#
Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, 0, a.Length + 1));
Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, a.Length + 1, 0));
Assert.Throws<ArgumentOutOfRangeException>(() => new Memory<int>(manager, 1, a.Length));
...
```

Reply:
Would have to Memory.Slice for the beyond array length as the .ctor doesn't validate (as it can't); so will add them that way

Reply:
What if we `CustomMemoryForTest.CreateMemory()`?

Reply:
Nevermind. Using CustomMemoryForTest would start testing that implementation (which doesn't help).

Reply:
The protected `CreateMemory` just calls the internal `Memory` .ctor; they both trust their inputs (other than making sure they are positive); as they have no additional data to validate against; which is why that `Memory` .ctor can only be accessed via the `protected` method inside `MemoryManager` - so they can't check if they are too big.

OTOH - the Memory.Slice should validate the default Memory method; which gets the length from `GetSpan` (which likewise provides the unvalidated `Span(void* pointer, int length)`; other than checking length, is positive as there is no additional data to validate against)

Reply:
i.e no one outside of a `MemoryManager` should be creating memory with a `MemoryManager`

Reply:
Can't have the `new Memory(MemoryManager, ...)` tests as they aren't available in the api surface